app/src/main/java/com/example/footballmanager/data/MatchesRepository.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.footballmanager.data
+
+import com.example.footballmanager.data.cache.MatchDao
+import com.example.footballmanager.data.entities.Match
+import com.example.footballmanager.data.network.FootballApiService
+import javax.inject.Inject
+
+class MatchesRepository @Inject constructor(
```

Comment:
Consider introducing a layer of abstraction eg. an interface for the Match data source. This would allow for easier swapping of implementations (e.g., different caching strategies, remote data sources).
Then you can use DI to [bind](https://www.valueof.io/blog/inject-provides-binds-dependencies-dagger-hilt) the implementation.